awswrangler/timestream.py

```diff
@@ -240,21 +249,22 @@ def write(
     _logger.debug("cols_names: %s", cols_names)
     batches: List[List[Any]] = _utils.chunkify(lst=_df2list(df=df[cols_names]), max_length=100)
     _logger.debug("len(batches): %s", len(batches))
-    with concurrent.futures.ThreadPoolExecutor(max_workers=num_threads) as executor:
-        res: List[List[Any]] = list(
-            executor.map(
-                _write_batch,
-                itertools.repeat(database),
-                itertools.repeat(table),
-                itertools.repeat(cols_names),
-                itertools.repeat(measure_cols_names),
-                itertools.repeat(measure_types),
-                itertools.repeat(version),
-                batches,
-                itertools.repeat(_utils.boto3_to_primitives(boto3_session=boto3_session)),
-            )
-        )
-        return [item for sublist in res for item in sublist]
+
+    executor = _get_executor(use_threads=use_threads)
+
+    res = executor.map(
+        _write_batch,
+        itertools.repeat(database),
```

Comment:
The second argument should be `boto3_session` like [here](https://github.com/awslabs/aws-data-wrangler/blob/release-3.0.0/awswrangler/lakeformation/_read.py#L69), this is because the new executor [logic](https://github.com/awslabs/aws-data-wrangler/blob/release-3.0.0/awswrangler/_threading.py#L31) is expecting the second argument to be a boto3 session. It was done to handle cases where the session must be broken down to primitives or in the case of Ray because it's not a pickable object